service/invoker/invoker.go

```diff
@@ -195,3 +162,40 @@ func (i *Invoker) ByHeightFrom(height uint64, header *flow.Header) (*flow.Header
 	// find the block in storage as all of them are guaranteed to be finalized
 	return i.index.Header(height)
 }
+
+func (i *Invoker) storageSnapshot(height uint64) snapshot.StorageSnapshot {
+	// Initialize the storage snapshot. We use a shared cache between all
+	// heights here. It's a smart cache, which means that items that are
+	// accessed often are more likely to be kept, regardless of height. This
+	// allows us to put an upper bound on total cache size while using it for
+	// all heights.
+	return snapshot.NewReadFuncStorageSnapshot(
+		readRegister(i.index, i.cache, height))
+}
+
+func readRegister(
+	index archive.Reader,
+	cache Cache,
+	height uint64,
+) func(flow.RegisterID) (flow.RegisterValue, error) {
+	return func(regID flow.RegisterID) (flow.RegisterValue, error) {
+		cacheKey := fmt.Sprintf("%d/%s", height, regID)
```

Comment:
we should [reuse this function and make it public](https://github.com/onflow/flow-archive/blob/master/service/storage2/payload/lookup.go#L18)

Reply:
not sure I understand. that function looks like it does something else...

Reply:
nvm, I think we can use a different key for the caching as long as its unique